src/lib/mathlib/math/Matrix.hpp

```diff
@@ -491,6 +491,49 @@ class __EXPORT Matrix<3, 3> : public MatrixBase<3, 3>
 
 		return euler;
 	}
+
+	/**
+	 * get euler angles from rotation matrix and avoid gimbal lock effects
+	 * euler.data[*] are the elements of the rotation matrix from body to nav frame (NED), i.e. Xn = Tbn * Xb
+	 * This routine checks for the Aircrafts flight regime and rotates the coordinate frame accordingly to avoid gimbal lock effects.
+	 * In plots you will notice a sudden 90° pitch angle change and roll/yaw axis of the body frame switch. Roll/Yaw in the NED framne remain the same.
+	 * coordinate_frame 0 = nose down
```

Comment:
I think we need a different architectural approach, because we can't be sure that someone is using euler angles in the system. We would need to completely strip euler angles from all messages to be safe, and do this conversion in the mavlink app.